libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -135,8 +135,16 @@ void AC_WPNav::wp_and_spline_init()
     // initialise feed forward velocity to zero
     _pos_control.set_desired_velocity_xy(0.0f, 0.0f);
 
+    // initialize the desired wp speed if not already done
+    _wp_desired_speed_cms = _wp_speed_cms;
+
+    // initialise the current wp speed to current speed along the track
+    const Vector3f &curr_vel = _inav.get_velocity();
+    // get speed along track (note: we convert vertical speed into horizontal speed equivalent)
+    _wp_current_speed_cms = curr_vel.x * _pos_delta_unit.x + curr_vel.y * _pos_delta_unit.y;
```

Comment:
We don't have a valid _pos_delta_unit at the time that wp_and_spline_init() is called so I think this is using the previous track to initialise the _wp_current_speed_cms...